### PR TITLE
Automatically Enable/disable depenencies tab

### LINKED ISSUE #163: Enable/disable dependencies tab in UI when dependency job is enabled/disabled

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #163

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/311)

File: pkg/strategy/controller_test.go [hunk 1]
```diff
@@ -246,38 +246,94 @@ func TestNormalizeUI(t *testing.T) {
...
-			j:        &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es.archive.enabled": "false"})}},
-			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es.archive.enabled": "false"})}},
+			j:        &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Type: "memory"}},
+			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Type: "memory"}},
 		},
 		{
 			j: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es-archive.enabled": "true"})}},
 			expected: &v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(map[string]interface{}{"es-archive.enabled": "true"})},
-				UI: v1.JaegerUISpec{Options: v1.NewFreeForm(map[string]interface{}{"archiveEnabled": true})}},
+				UI: v1.JaegerUISpec{Options: v1.NewFreeForm(map[string]interface{}{"archiveEnabled": true, "dependencies": map[string]interface{}{"menuEnabled": false}})}},
```
Comment (objectiser) [inline pkg/strategy/controller_test.go]: Isn't this wrong - if the storage type is not defined, it should default to `memory`, which should show the dependencies tab. The expected results is showing it disabled.

File: pkg/strategy/controller_test.go [hunk 2]
```diff
@@ -246,38 +246,94 @@ func TestNormalizeUI(t *testing.T) {
...
+		{
+			uiOpts:   map[string]interface{}{"dependencies": map[string]interface{}{"menuEnabled": "respectThis"}},
+			storage:  "whateverStorage",
+			expected: map[string]interface{}{"dependencies": map[string]interface{}{"menuEnabled": "respectThis"}},
+		},
+		{
+			uiOpts:   map[string]interface{}{"dependencies": map[string]interface{}{"foo": "bar"}},
+			storage:  "whateverStorage",
+			expected: map[string]interface{}{"dependencies": map[string]interface{}{"foo": "bar", "menuEnabled": false}},
+		},
```
Comment (objectiser) [inline pkg/strategy/controller_test.go]: Can you add one test with `test.enabled` set to `false`.

File: pkg/strategy/controller_test.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/strategy/controller_test.go]: In general yes, but this tests only normalizeUI which works on any storage

File: pkg/strategy/controller.go [hunk 3]
```diff
@@ -135,23 +135,48 @@ func normalizeRollover(spec *v1.JaegerEsRolloverSpec) {
...
+		if strings.EqualFold(sOpts["es-archive.enabled"], "true") ||
+			strings.EqualFold(sOpts["cassandra-archive.enabled"], "true") {
+			uiOpts["archiveEnabled"] = true
+		}
+	}
+}
+
+func disableDependenciesTab(uiOpts map[string]interface{}, storage string, depsEnabled *bool) {
+	// dependency tab is by default enabled and memory storage support it
+	if strings.EqualFold(storage, "memory") || (depsEnabled != nil && *depsEnabled == false) {
```
Comment (objectiser) [inline pkg/strategy/controller.go]: Just realised, with the additional test, if dependencies are explicitly disabled, then menuEnabled should be false?

Comment (objectiser) [reply pkg/strategy/controller.go]: so should the condition be `if strings.EqualFold(storage, "memory") || depsEnabled == nil || *depsEnabled == true) {`?

Comment (pavolloffay) [reply pkg/strategy/controller.go]: You are right, changing

File: pkg/strategy/controller.go [hunk 4]
```diff
@@ -161,7 +161,7 @@ func enableArchiveButton(uiOpts map[string]interface{}, sOpts map[string]string)
 
 func disableDependenciesTab(uiOpts map[string]interface{}, storage string, depsEnabled *bool) {
 	// dependency tab is by default enabled and memory storage support it
-	if strings.EqualFold(storage, "memory") || (depsEnabled != nil && *depsEnabled == false) {
+	if strings.EqualFold(storage, "memory") || (depsEnabled != nil && *depsEnabled == true) {
```
Comment (objectiser) [inline pkg/strategy/controller.go]: I think it needs to be `... || depsEnabled == nil || *depsEnabled == true` - as if the `depsEnabled` is nil, then it means dependencies are enabled by default, doesn't it?

Comment (pavolloffay) [reply pkg/strategy/controller.go]: The normalization of deps spec runs before - so we assuming correct settings here

**Comment by objectiser** [comment]
@pavolloffay LGTM - although there is a test failure.

**Pull request merged**